src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx

```diff
@@ -36,7 +36,9 @@ export const ActivityFlowBuilder = () => {
   const { control, watch } = useFormContext();
   const { appletId, activityFlowId } = useParams();
   const activityFlows: AppletFormValues['activityFlows'] = watch('activityFlows');
-  const activityFlowIndex = activityFlows.findIndex((flow) => flow.id === activityFlowId);
+  const activityFlowIndex = activityFlows.findIndex(
+    (flow) => (flow.id || flow.key) === activityFlowId,
```

Comment:
could you move this function and the same function in ActivityFlowAbout.tsx, maybe make it more generic:
```
const checkOnIdOrKey = (checkId: string) => (entity: TypeThatHasIdAndKey) => (entity.id || entity.key) === checkId
...
const activityFlowIndex = activityFlows.findIndex(checkIdOrKey(activityOnFlowId));
```

Reply:
done